src/xdocs/checks/coding/whenshouldbeused.xml

```diff
@@ -0,0 +1,122 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<document xmlns="http://maven.apache.org/XDOC/2.0"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/XDOC/2.0 https://maven.apache.org/xsd/xdoc-2.0.xsd">
+  <head>
+    <title>WhenShouldBeUsed</title>
+  </head>
+  <body>
+    <section name="WhenShouldBeUsed">
+      <p>Since Checkstyle 10.18.0</p>
+      <subsection name="Description" id="Description">
+        <p>
+          Ensures that <code>when</code> is used instead of a single
+          <code>if</code> statement inside a case block.
+        </p>
+        <p>
+          Java 21 has introduced enhancements for switch statements and expressions
```

Comment:
Isn't this a `Rationale`?

Reply:
I don't know I felt like this is a general explanation for the syntax and the `Rationale` is the last scentence only in this paragragh. Do you think we should add a `Rationale:` ?

Reply:
I think so, Yes. You are explaining the general syntax but it helps lead into the real rationale if they didn't know what the syntax was.

Reply:
done

Reply:
@mahfouz72 please always share a link to created issues: https://github.com/checkstyle/checkstyle/issues/15270